src/shared/firewall-util.c

```diff
@@ -40,76 +65,69 @@ FirewallContext *fw_ctx_free(FirewallContext *ctx) {
         if (!ctx)
                 return NULL;
 
-        if (ctx->firewall_backend == FW_BACKEND_NFTABLES)
+        if (ctx->backend == FW_BACKEND_NFTABLES)
                 fw_nftables_exit(ctx);
 
         return mfree(ctx);
 }
 
 int fw_add_masquerade(
-                FirewallContext **fw_ctx,
+                FirewallContext **ctx,
                 bool add,
                 int af,
                 const union in_addr_union *source,
                 unsigned source_prefixlen) {
-        FirewallContext *ctx;
-        int r;
```

Comment:
I'd alway grant `int r;` the special privilege of always being declared out the outermost function scope. It's a bit a special variable for us, it's where we write function return values too in almost all cases. It's not really like any other variable hence, it' *the* variable for doing error handling, and I'd say that gives is the special right to appear always in the topmost function scope. Hence I'd leave this line here...

(But of course, this doesn't matter at all)